### PR TITLE
Feat: article

### PR DESCRIPTION
**Pull request by sericaia**
TODO:
- fix markup paragraphs which no longer work like before

### CODE REVIEW

File: packages/article/article-meta.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { View } from "react-native";
+
+import ArticleByline from "@times-components/article-byline";
+import DatePublication from "@times-components/date-publication";
+
+import styles from "./article-style";
+
+const ArticleHeader = ({ byline, publishedTime, publicationName }) =>
```
Comment (sericaia) [inline packages/article/article-meta.js]: should this file be renamed to article-header?

File: packages/article/article-style.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import { StyleSheet, Platform } from "react-native";
+
+const standFirstFontFamily = "TimesModern-Regular";
+const bodyFontFamilyWebIos = "TimesDigitalW04";
+const bodyFontFamilyAndroid = "TimesDigitalW04-Regular";
+const borderColor = "#d0cece";
+const borderWidth = StyleSheet.hairlineWidth;
+
+const globalStyle = StyleSheet.create({
+  Container: {
```
Comment (sericaia) [inline packages/article/article-style.js]: Shouldn't we use camelCase in this styles?

File: packages/article/article.js [hunk 3]
```diff
@@ -1,11 +1,153 @@
...
+      get(articleData, "content", []).map(i => ({
+        type: "article_body_row",
+        data: i
+      }))
+    );
+    return ArticleArray;
+  }
+
+  static renderRow(rowData) {
+    if (rowData.type === "ads" && Platform.OS === "web") {
```
Comment (sericaia) [inline packages/article/article.js]: this "ads" part should be removed for now, right? and afaik it will use what @thatportugueseguy @JGAntunes are doing

Comment (nehasri89) [reply packages/article/article.js]: actually, those guys are doing the ads within the article and this is the header ad, so this stays but need review from @thatportugueseguy and @JGAntunes as discussed in yesterday's standup

File: packages/article/article.js [hunk 4]
```diff
@@ -1,11 +1,153 @@
...
+      )
+    };
+    return (
+      <View style={styles.PageWrapper}>
+        <ListView
+          dataSource={this.state.dataSource}
+          renderRow={ArticlePage.renderRow}
+          initialListSize={10}
+          scrollRenderAheadDistance={10}
+          pageSize={1}
```
Comment (sericaia) [inline packages/article/article.js]: should we move this config values somewhere?

File: packages/article/article.js [hunk 5]
```diff
@@ -1,11 +1,153 @@
...
+ArticlePage.propTypes = {
+  data: PropTypes.shape({
+    article: PropTypes.object,
+    loading: PropTypes.boolean
+  })
+};
+
+ArticlePage.defaultProps = {
+  data: {
+    loading: true
```
Comment (sericaia) [inline packages/article/article.js]: if `article` is not required, we need to add a default prop

File: packages/article/article.js (hunk 3 above)
Comment (nehasri89) [reply packages/article/article.js]: confirmed with @craigbilner remove the ad from here and will raise a separate story

File: packages/article/article.js [hunk 6]
```diff
@@ -1,11 +1,153 @@
...
+  render() {
+    if (this.props.data.loading) {
+      return (
+        <View style={styles.Container}>
+          <ActivityIndicator size={"large"} />
+        </View>
+      );
+    }
+
+    this.state = {
```
Comment (sericaia) [inline packages/article/article.js]: I'm really confused about this. We should not mutate state here and every time we mutate state we should use`this.setState` and not mutating it directly. The only case where we can assign to `this.state` is in the constructor

File: packages/article/article.js [hunk 7]
```diff
@@ -1,11 +1,153 @@
...
+    const ArticleArray = Array.prototype.concat(
+      [
+        { type: "ads", data: adsData },
+        { type: "leadAsset", data: leadAssetData },
+        { type: "header", data: articleHeaderData },
+        { type: "middleContainer", data: articleMidContainerData }
+      ],
+      // NOTE failing gracefully
+      get(articleData, "content", []).map(i => ({
+        type: "article_body_row",
```
Comment (sericaia) [inline packages/article/article.js]: do we really need this?

File: packages/article/article.js (hunk 6 above)
Comment (sericaia) [reply packages/article/article.js]: consider https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops

File: packages/article/article.js [hunk 8]
```diff
@@ -1,11 +1,146 @@
...
+      //       >
+      //         {React.cloneElement(el, {
+      //           style: StyleSheet.flatten([styles.articleTextElement])
+      //         })}
+      //       </View>
+      //     )}
+      //   </View>
+      // );
+      // TODO :: fix this, builder does not exit now
+      return null;
```
Comment (sericaia) [inline packages/article/article.js]: this return will never be used

File: packages/article/CHANGELOG.md [hunk 9]
```diff
@@ -3,35 +3,6 @@
 All notable changes to this project will be documented in this file.
 See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
 
-<a name="0.5.3"></a>
```
Comment (JGAntunes) [inline packages/article/CHANGELOG.md]: Should this PR be interfering with the CHANGELOG like this? (removing releases)

File: packages/article/article.js [hunk 10]
```diff
@@ -1,11 +1,146 @@
...
+    } else if (rowData.type === "article_body_row") {
+      // return (
+      //   <View style={[styles.articleMainContentRow, styles.articleText]}>
+      //     {builder({ ast: [rowData.data] }).map(el =>
+      //       <View
+      //         style={styles.articleTextWrapper}
+      //         key={`paragraph-${Date.now().toLocaleString()}`}
+      //       >
+      //         {React.cloneElement(el, {
+      //           style: StyleSheet.flatten([styles.articleTextElement])
```
Comment (sericaia) [inline packages/article/article.js]: why do you need StyleSheet.flatten @killanaca ?

File: packages/article/article.js (hunk 8 above)
Comment (nehasri89) [reply packages/article/article.js]: yes, but eslint complains of not having a return from a method, so either we can disable eslint for this block or have a null return

**Comment by killanaca** [comment]
#234

**Pull request closed without merging**